crowbar_framework/app/models/crowbar_service.rb

```diff
@@ -152,6 +152,13 @@ def transition(inst, name, state)
       if node.crowbar["state"] != state
         @logger.debug("Crowbar transition: state has changed so we need to do stuff for #{name} to #{state}")
 
+        # Do not allow change to shutdown state from crowbar_upgrade
+        # (we need to reboot the nodes for upgrade, but without changing the state)
+        if node.crowbar["state"] == "crowbar_upgrade" && state == "shutdown"
+          @logger.debug("Crowbar transition: current state is crowbar_upgrade; change not allowed")
```

Comment:
Could you tweak the log message to
 'current node state is "crowbar_upgrade" ignoring change to "shutdown"'